lib/sourceParams.js

```diff
@@ -374,8 +374,8 @@ module.exports = (params, app, appSchema) => {
       for (const paramKey in paramSet) {
         const param = paramSet[paramKey]
 
-        // recurse if param value is an object
-        if (param === Object(param)) {
+        // recurse if param value is an object (but not a function)
+        if (param === Object(param) && typeof param !== 'function') {
```

Comment:
Added this change as it hit a function like `onServerStart` in the params and tried parsing it which caused an infinite loop and threw an error for max callstack reached.